### DISCUSSION TITLE
NCBI databaseについて

### DISCUSSION THREAD

**Discussion by ShoheiOgino** [nig-sc/nigsc_homepage2#467] [Q&A] [answered]
お忙しいところすみません.
ncbiのdatabaseをblastのdatabaseとして使用したいと考えております.
Singularityイメージではなくインストールしたblastを使用しております.
/usr/local/seq/blast/ncbi/nrにblastnにて使用できるncbiのdatabaseがあるとお伺いしているのですが,
blastn -db /usr/local/seq/blast/ncbi/nr -query queryname.faと指定しても
BLAST Database error: No alias or index file found for nucleotide database [/usr/local/seq/blast/ncbi/nr] in search path
と表示されてしまいます.
どのようにすればよろしいでしょうか.

お手数をおかけして大変申し訳ございませんがご教示いただけますと幸いです.
何卒よろしくお願いいたします.

**Comment by sc-fs** [comment]
/usr/local/seq/blast/ncbi は、ログインノードより先の計算機に存在しています。
実行されたqsubのスクリプト及びＪＯＢ番号を教えてください。

**Comment by ShoheiOgino** [comment]
ご返信ありがとうございます。

JOB番号は25925581となります。
スクリプトは下記の通りです。singularityのblastとダウンロードしたblastを両方走らせてみております。

```shell
#!/usr/bin/bash
#$ -S /usr/bin/bash
#$ -cwd
#$ -V
#$ -l short
#$ -l s_vmem=4G
#$ -l mem_req=4G
#$ -N blastdata


blastn -db /usr/local/seq/blast/ncbi/nr -query ../../test.fasta -out blastout.txt

singularity exec -e --bind /usr/local/seq/blast/ncbi /usr/local/biotools/b/blast:2.7.1--boost1.64_1 blastn -query ../../test.fasta -db /usr/local/seq/blast/ncbi/nr
``` 


お忙しいところお手数をおかけし大変申し訳ございませんが,何卒よろしくお願いいたします。

**Comment by sc-fs** [comment, marked as answer]
指定されていたDBの種類は、
/usr/local/seq/blast/ncbi/nr は　Protein
起動するコマンド：blastnは塩基配列の検索ですので、
DBとコマンドが合っていないのが原因です。

/usr/local/seq/blast/ncbi/nt はNucleotide
ですので、こちらをDBとして利用することになります。

なお、どちらを実行する予定なのか不明ですが、それぞれ、以下のようになります。
（blastはインストールされている最新としています）
```
#!/usr/bin/bash
#$ -S /usr/bin/bash
#$ -cwd
#$ -V
#$ -l s_vmem=384G
#$ -l mem_req=384G
#$ -N blastdata

# cmd:blastp  DB:/usr/local/seq/blast/ncbi/nr Protein
singularity exec --bind /usr/local/seq /usr/local/biotools/b/blast:2.15.0--pl5321h6f7f691_1 blastp -db /usr/local/seq/blast/ncbi/nr -query test.fasta -out blastp.txt
# cmd:blastn  DB:/usr/local/seq/blast/ncbi/nt Nucleotide
singularity exec --bind /usr/local/seq /usr/local/biotools/b/blast:2.15.0--pl5321h6f7f691_1 blastn -db /usr/local/seq/blast/ncbi/nt -query test.fasta -out blastn.txt
```

**Reply by ShoheiOgino** [reply]
ご丁寧にありがとうございます。
無事に実行することができました。

お忙しい中お手数をおかけして大変申し訳ございませんだした。
ご対応ありがとうございました。
